### PR TITLE
Modal keyboarding fixes

### PR DESCRIPTION
**Pull request by SavannahBourgeois**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This change fixes two bugs we have in office-online-ui related to dialogs and keyboarding. 

**Issue 1:** If focus is inside of a spin button and the dialog is in keyboard move mode, the spin button value updates when the user presses "up" or "down" to position the dialog.
**Fix:** Instead of an onKeyDown event on the dialog, I added a global onKeyDown event when we are in isInKeyboardMoveMode. This way the keyboard events will go to this caller before any other component. Now, the spin button inside of the dialog will not get keyboard events that are intended for moving the dialog.

**Issue 2:** If focus is inside of a combobox within a dialog and the user hits the shortcut key (ctrl+alt+space) to open the dialog move menu, nothing happens and the menu does not open.
**Fix:** Similar to Issue #1, we added another global handler for onKeyUp instead of setting it on the focus trap zone. This way the keyboard events will go to this caller before any other component. We additionally check if the keyboard event was the shortcut key AND if the target was a child of the dialog content. Now, the dialog move menu will open when you do the shortcut with the focus inside of a combobox.

Additionally to help verify these changes in the future, I added the drag options to the sample blocking dialog and added a spin button and combobox inside. I have verified that the issues are fixed and the menu opens as expected in Firefox since the code mentions it as a special use case. I have also verified with modeless dialogs, if the focus is not within the dialog that hitting the keyboard shortcut does not open the menu. It does when focus is within the dialog contents.

#### Focus areas to test

Dialogs and modals.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/11042)

### CODE REVIEW

**Comment by SavannahBourgeois** [comment]
@jspurlin here are the changes for dialog keyboard we found yesterday

**Comment by SavannahBourgeois** [comment]
@JasonGore can you take a look?

**Comment by JasonGore** [comment]
You mentioned @jspurlin so I was hoping he would review. It sounds like he has more context. Thanks!

**Comment by SavannahBourgeois** [comment]
Unfortunately Jeremy is on vacation all week. I talked over the idea with him before he left though to make sure he agreed with this way to work around it.

File: packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx [hunk 1]
```diff
@@ -181,8 +185,6 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
         forceFocusInsideTrap={isModeless ? !isModeless : forceFocusInsideTrap}
         firstFocusableSelector={firstFocusableSelector}
         focusPreviouslyFocusedInnerElement={true}
-        onKeyDown={dragOptions ? this._onDialogKeyDown : undefined}
```
[the remark this answers is not in the document]
Comment (SavannahBourgeois) [reply packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx]: Yes. It is only needed when we are in "move" mode which dragOptions adds the support for. If we do not allow the dialog to be moved, then we do not need the onKeyDown event.

File: packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx [hunk 2]
```diff
@@ -282,6 +284,10 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
       y: 0
     });
 
+    if (this.props.dragOptions) {
+      this._events.off(window, 'keyup', this._onKeyUp, true /* useCapture */);
+    }
```
[the remark this answers is not in the document]
Comment (SavannahBourgeois) [reply packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx]: Do you prefer if I add a Boolean that says "we registered" or something like that an unregister based off of that instead?

[the remark this answers is not in the document]
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx]: I think it's ok as is since `BaseComponent` will handle cleanup.

**Pull request merged**